packages/typescript-estree/src/create-program/getParsedConfigFile.ts

```diff
@@ -33,7 +33,11 @@ function getParsedConfigFile(
       fileExists: fs.existsSync,
       getCurrentDirectory,
       readDirectory: tsserver.sys.readDirectory,
-      readFile: file => fs.readFileSync(file, 'utf-8'),
+      readFile: file =>
+        fs.readFileSync(
+          path.isAbsolute(file) ? file : path.join(getCurrentDirectory(), file),
+          'utf-8',
+        ),
```

Comment:
Without this absolute checking:

```plaintext
      A fatal parsing error occurred: Parsing error: Could not read project service default project 'tsconfig.json': error TS5012: Cannot read file '/home/runner/work/typescript-eslint/typescript-eslint/packages/eslint-plugin-internal/tests/fixtures/tsconfig.build.json': ENOENT: no such file or directory, open '/home/runner/work/typescript-eslint/typescript-eslint/packages/eslint-plugin-internal/tests/fixtures/tsconfig.build.json'.
```

That `tsconfig.build.json` comes from `packages/typescript-estree/tsconfig.json`. The test was reading with cwd `packages/typescript-estree`.